pydmd/costs.py

```diff
@@ -0,0 +1,1457 @@
+        elif not self._global_svd and self._svd_rank > 0:
+            if self._force_even_eigs and self._svd_rank % 2:
+                raise ValueError(
+                    "svd_rank is odd, but force_even_eigs is True."
+                )
+            if self._svd_rank > self._n_data_vars:
+                raise ValueError(
+                    "Rank is larger than the data spatial dimension."
+                )
+            svd_rank_pre_allocate = compute_rank(data, svd_rank=self._svd_rank)
+        # If not using a global svd or a specified svd_rank, local u from each window is
+        # used instead. The optimal svd_rank may change when using the locally optimal
+        # svd_rank. To deal with this situation in the pre-allocation we give the
+        # maximally allowed svd_rank for pre-allocation.
+        elif self._max_rank is not None:
+            svd_rank_pre_allocate = self._max_rank
+        else:
+            svd_rank_pre_allocate = self._n_data_vars
+
+        # Pre-allocate all elements for the sliding window DMD.
+        self._time_array = np.zeros((self._n_slides, self._window_length))
+        self._modes_array = np.zeros(
+            (self._n_slides, self._n_data_vars, svd_rank_pre_allocate),
+            np.complex128,
+        )
+        self._omega_array = np.zeros(
+            (self._n_slides, svd_rank_pre_allocate), np.complex128
+        )
+        self._amplitudes_array = np.zeros(
+            (self._n_slides, svd_rank_pre_allocate), np.complex128
+        )
+        self._window_means_array = np.zeros((self._n_slides, self._n_data_vars))
+
+        # Get initial values for the eigenvalues.
+        self._init_alpha = self._build_initizialization()
+
+        # Initialize the BOPDMD object.
+        optdmd = BOPDMD(
+            svd_rank=self._svd_rank,
+            init_alpha=self._init_alpha,
+            **self._pydmd_kwargs,
+        )
+
+        # Round the corners of the windowed data towards zero, which shrinks
+        # the real components of the fitted eigenvalues away from unrealistic
+        # exponential growth.
+        lv_kern = self.calculate_lv_kern(
+            self._window_length, corner_sharpness=corner_sharpness
+        )
+
+        # Perform the sliding window DMD fitting.
+        for k in range(self._n_slides):
+            if verbose:
+                if k // 50 == k / 50:
+                    print("{} of {}".format(k, self._n_slides))
+
+            sample_slice = self.get_window_indices(k)
+            data_window = data[:, sample_slice]
+            original_time_window = time[:, sample_slice]
+
+            # All windows are fit with the time array reset to start at t=0.
+            t_start = original_time_window[:, 0]
+            time_window = original_time_window - t_start
+
+            # Subtract off the time mean before rounding corners.
+            c = np.mean(data_window, 1, keepdims=True)
+            data_window = data_window - c
```

Comment:
```suggestion
            data_window -= np.mean(data_window, 1, keepdims=True)
```
Double check that the result is the same because you are using the same variable on both side. The main thing is that `c` is not needed.